File: FreStyle/src/main/java/com/example/FreStyle/entity/ConversationTemplate.java
package com.example.FreStyle.entity;

import java.sql.Timestamp;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "conversation_templates")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ConversationTemplate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "title", length = 100, nullable = false)
    private String title;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "category", length = 50, nullable = false)
    private String category;

    @Column(name = "opening_message", columnDefinition = "TEXT", nullable = false)
    private String openingMessage;

    @Column(name = "system_context", columnDefinition = "TEXT", nullable = false)
    private String systemContext;

    @Column(name = "difficulty", length = 20)
    private String difficulty;

    @Column(name = "created_at", insertable = false, updatable = false)
    private Timestamp createdAt;
}

File: backend/internal/domain/conversation_template.go
package domain

import "time"

type ConversationTemplate struct {
	ID        uint64    `gorm:"primaryKey" json:"id"`
	Title     string    `gorm:"column:title" json:"title"`
	Body      string    `gorm:"column:body" json:"body"`
	Category  string    `gorm:"column:category" json:"category"`
	IsActive  bool      `gorm:"column:is_active" json:"isActive"`
	CreatedAt time.Time `gorm:"column:created_at" json:"createdAt"`
}

func (ConversationTemplate) TableName() string { return "conversation_templates" }
